PokeAlarm/Alarms/Pushbullet/PushBulletAlarm.py

```diff
@@ -182,19 +184,59 @@ def invasion_alert(self, invasion_info):
     # Attempt to get the channel, otherwise default to all devices
     def get_sender(self, channel_tag):
         req_channel = next(
-            (channel for channel in self.__client.channels
-             if channel.channel_tag == channel_tag), self.__client)
-        if req_channel is self.__client and channel_tag is not None:
+            (channel for channel in self.__channels
+             if channel['tag'] == channel_tag), None)
+        if req_channel is None and channel_tag is not None:
             self._log.error(
-                "Unable to find channel.Pushing to all devices instead.")
+                "Unable to find channel. Pushing to all devices instead.")
         else:
             self._log.debug("Setting to channel %s." % channel_tag)
         return req_channel
 
     # Push a link to the given channel
     def push_link(self, sender, title, url, body):
-        sender.push_link(title=title, url=url, body=body)
+        data = {"type": "link", "title": title, "url": url, "body": body}
+        if sender is not None:
+            data.update({'channel_tag': sender['tag']})
+        self.push(data)
 
     # Push a link to the given channel
     def push_note(self, sender, title, message):
-        sender.push_note(title, message)
+        data = {"type": "note", "title": title, "body": message}
+        if sender is not None:
+            data.update({'channel_tag': sender['tag']})
+        self.push(data)
+
+    def push(self, data):
+        res = requests.post(
+            'https://api.pushbullet.com/v2/pushes',
+            headers={
+                'Access-Token': self.__api_key,
+                'Content-Type': 'application/json'
+            },
+            data=json.dumps(data))
+        if res.status_code == requests.codes.ok:
+            self._log.debug('Notification successful '
+                            f'(returned {res.status_code})')
+        else:
+            self._log.debug(f'PushBullet response was {res.content}')
+            raise requests.exceptions.RequestException(
+                f'Response received {res.status_code}, webhook not accepted.')
+
+    def update_channels(self):
+        self.__channels = {}
+        response = requests.get(
+            'https://api.pushbullet.com/v2/channels',
+            headers={
+                'Access-Token': self.__api_key
+            }, timeout=30)
+        if response.ok is True:
+            self.__channels = response.json()['channels']
+            self._log.debug('Detected the following PushBullet channels: {}'
+                            .format(self.__channels))
+        else:
```

Comment:
It's better to avoid branching the decision tree here. If you do:
```python
if not response.ok:
    // handle error

//handle normally
```

This is easier for people to read because it reinforces the non-error path as "default" - and error handling is more of a "if something goes wrong, handle it".